src/main/java/com/tailf/jnc/Device.java

```diff
@@ -150,6 +166,48 @@ public Device(String name, String mgmtIp, int mgmtPort) {
         trees = new ArrayList<SessionTree>();
     }
 
+    /**
+     * Builder for Device instances using the NETCONF call-home
+     * mechanism.
+     */
+    public static class CallHome {
+        private String name;
+        private int waitTimeout = 0;
+        private int kexTimeout = 0;
+        private String knownHostsFile = null;
+        private int port = 4334;
+
+        public CallHome(String deviceName) {
+            name = deviceName;
+        }
+        public CallHome setWaitTimeout(int waitTimeout) {
+            this.waitTimeout = waitTimeout;
+            return this;
+        }
+        public CallHome setKexTimeout(int kexTimeout) {
+            this.kexTimeout = kexTimeout;
+            return this;
+        }
+        public CallHome setVerify() {
+            return setKnownHostsFile("~/.ssh/known_hosts");
+        }
+        public CallHome setKnownHostsFile(String knownHostsFile) {
+            this.knownHostsFile = knownHostsFile;
+            return this;
+        }
+        public CallHome setPort(int port) {
```

Comment:
could various devices managed by this here need/want to use different ports?

Reply:
The 4334 port is the standard one, but there's no reason why it would have to be the only one; other clients / client libraries allow other ports too.

Reply:
i got confused a bit an thought this is "server part" port config, and question was in regard to allowing various ports per client...